spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/KerasLayer.scala

```diff
@@ -51,23 +67,94 @@ object KerasLayerSerializer extends ContainerSerializable with TKerasSerializerH
   }
 }
 
+private[bigdl] class KerasIdentityActivationWrapper[T: ClassTag]
+(val torchLayer: AbstractModule[Activity, Activity, T],
+    val activation: KerasLayer[Tensor[T], Tensor[T], T])(implicit ev: TensorNumeric[T])
+  extends KerasLayer[Activity, Activity, T](null) {
+  require(!torchLayer.isKerasStyle(),
+    s"We should get torch style layer, but got: $torchLayer")
+
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    torchLayer.computeOutputShape(inputShape)
+  }
+
+  override def doBuild(inputShape: Shape): AbstractModule[Activity, Activity, T] = {
+    val torchActivation = activation.doBuild(inputShape)
+    require(!torchActivation.isKerasStyle(),
+      s"We should get torch style activation, but got: $torchActivation")
+    val seq = TSequential[T]()
+    seq.add(torchLayer)
+    seq.add(torchActivation)
+    seq.setName(torchLayer.getName())
+    seq
+  }
+}
+
+/**
+ * Wrap a torch style layer to keras style layer.
+ * This layer can be built multiple times.
+ * We are supposing the inputshape and the outputshape keep the same in this layer.
+ * @param layer a torch style layer
+ * @return a keras compatible layer
+ */
+class KerasIdentityWrapper[T: ClassTag]
+(val layer: AbstractModule[Activity, Activity, T])(implicit ev: TensorNumeric[T])
+  extends KerasLayer[Activity, Activity, T](null) {
+  if (layer.isKerasStyle()) {
+    throw new RuntimeException(s"We only accept torch layer here, but got: $layer")
+  }
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    inputShape
+  }
+  override def doBuild(inputShape: Shape): AbstractModule[Activity, Activity, T] = layer
+
+  override def allowRebuilt(): Boolean = true
+
+}
+
+/**
+ * Wrap a torch style layer to keras style layer.
+ * This layer can be built multiple times.
+ * @param torchLayer a torch style layer
+ *   i.e If the input data is (2, 3, 4) and 2 is the batch size, you should input: (3, 4) here.
+ * @return a keras compatible layer
+ */
+class KerasLayerWrapper[T: ClassTag]
+(val torchLayer: AbstractModule[Activity, Activity, T],
+    val inputShape: Shape = null)(implicit ev: TensorNumeric[T])
+  extends KerasLayer[Activity, Activity, T](KerasLayer.addBatch(inputShape)) {
+
+  require(!torchLayer.isKerasStyle(), s"We only accept torch layer here, but got: $torchLayer")
+
+  override def computeOutputShape(calcInputShape: Shape): Shape = {
+    val dummyOutTensor =
+      torchLayer.forward(Tensor[T](
+        (List(2) ++ KerasLayer.removeBatch(calcInputShape).toSingle()).toArray).rand())
+    val outSize = dummyOutTensor.toTensor.size()
+    KerasLayer.addBatch(Shape(outSize.slice(1, outSize.length)))
+  }
+
+  override def doBuild(inputShape: Shape): AbstractModule[Activity, Activity, T] = torchLayer
+
+  override def allowRebuilt(): Boolean = true
```

Comment:
Why do we need to support rebuilt here?

Reply:
There's no overhead for rebuilding this layer.

Reply:
`KerasLayerWrapper`  is used to combine activity and layer and it wold be built within `doBuild` method, so we need to enable `allowRebuilt` here.